rts/motoko-rts/src/memory/ic.rs

```diff
@@ -24,13 +24,18 @@ pub(crate) static mut LAST_HP: u32 = 0;
 
 // Provided by generated code
 extern "C" {
-    pub(crate) fn get_heap_base() -> u32;
+    fn get_heap_base() -> u32;
     pub(crate) fn get_static_roots() -> Value;
 }
 
+pub(crate) unsafe fn get_aligned_heap_base() -> u32 {
+    // align to 32 bytes
+    (get_heap_base() + 31) / 32 * 32
+}
```

Comment:
Again, I won't be maintaining this code so feel free to ignore, but it will be very confusing to readers why we need to align the heap base, so documentation here would be good.

Yesterday we talked about adding one more byte to the bitmap when heap base is not aligned, did that not work out? Why do you choose to align heap base instead of adding one more byte to the bitmap when it's not already aligned?

Reply:
Documentation is coming.

It did not work out, see https://github.com/dfinity/motoko/pull/2927#issuecomment-977006471.

Reply:
PTAL: ea18318